src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2219,6 +2220,11 @@ protected void runMayThrow()
                     handler.onFailure(FBUtilities.getBroadcastAddressAndPort(), RequestFailureReason.READ_TOO_MANY_TOMBSTONES);
                     logger.error(t.getMessage());
                 }
+                else if (t instanceof QueryReferencingTooManyIndexesException)
+                {
+                    handler.onFailure(FBUtilities.getBroadcastAddressAndPort(), RequestFailureReason.READ_TOO_MANY_INDEXES);
+                    logger.error(t.getMessage());
+                }
```

Comment:
I *think* we don't have to do this. It was a special case for tombstone thresholds, which existed before the new framework that local read size et al use.